x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/model_select.tsx

```diff
@@ -9,25 +9,219 @@ import React from 'react';
+      )}
+    </EuiText>
+  </EuiPanel>
+);
+
+export const SelectedModel: React.FC<MlModel> = (model) => {
+  const { createModel, startModel } = useActions(ModelSelectLogic);
+  const { areActionButtonsDisabled } = useValues(ModelSelectLogic);
+
+  return (
+    <EuiPanel color="subdued" title="Selected model">
+      <EuiPanel>
+        <EuiFlexGroup direction="column" gutterSize="s">
+          <EuiFlexItem>
+            <EuiTitle size="xs">
+              <h4>{model.title}</h4>
+            </EuiTitle>
+          </EuiFlexItem>
+          <EuiFlexItem>
+            <EuiTextColor color="subdued">{model.modelId}</EuiTextColor>
+          </EuiFlexItem>
+          {model.description && (
+            <EuiFlexItem>
+              <EuiText size="xs">{model.description}</EuiText>
+            </EuiFlexItem>
+          )}
+          {model.licenseType && (
+            <EuiFlexItem grow={false}>
+              {/* Wrap in a div to prevent the badge from growing to a whole row on mobile */}
+              <div>
+                <LicenseBadge
+                  licenseType={model.licenseType}
+                  modelDetailsPageUrl={model.modelDetailsPageUrl}
+                />
+              </div>
+            </EuiFlexItem>
+          )}
+          {(model.isPlaceholder ||
+            [
+              MlModelDeploymentState.Downloading,
+              MlModelDeploymentState.NotDeployed,
+              MlModelDeploymentState.Starting,
+            ].includes(model.deploymentState)) && (
+            <>
+              <EuiHorizontalRule margin="xs" />
+              <EuiFlexItem grow={false}>
+                <EuiFlexGroup alignItems="center" gutterSize="s">
+                  {model.isPlaceholder ? (
+                    <>
+                      <EuiFlexItem grow={false}>
+                        <DeployModelButton
+                          onClick={() => createModel(model.modelId)}
+                          disabled={areActionButtonsDisabled}
+                        />
+                      </EuiFlexItem>
+                      <EuiFlexItem>
+                        <EuiText size="xs">
+                          <p>
+                            <EuiTextColor color="danger">
+                              {i18n.translate(
+                                'xpack.enterpriseSearch.content.indices.pipelines.modelSelect.modelNotDeployedError',
+                                { defaultMessage: 'Model must be deployed before use.' }
+                              )}
+                            </EuiTextColor>
+                          </p>
+                        </EuiText>
+                      </EuiFlexItem>
+                    </>
+                  ) : model.deploymentState === MlModelDeploymentState.Downloading ? (
+                    <EuiFlexItem grow={false}>
+                      <ModelDeployingButton />
+                    </EuiFlexItem>
+                  ) : model.deploymentState === MlModelDeploymentState.NotDeployed ? (
```

Comment:
Why aren't we using `MlModelDeploymentState.Downloaded` to trigger this state?

Reply:
We're extending the Start functionality to all models, not just ELSER and E5. 3rd party models don't have a `Downloaded` state; they can be started when they are `NotDeployed`.

We _could_ modify this condition to be `Downloaded || NotDeployed`, but there's still the status badge that would show different labels depending on the model type. So it's easier to set the curated models' state to `NotDeployed` after being downloaded and manage that single use case.

Reply:
Ok, I think I understand, but it's easy to get confused with `MlModelDeploymentState` values getting remapped based on the model that it applies to. Can you add comments explaining the remapping wherever you set or handle an `MlModelDeploymentState` value that is not indicative of the model's true state, such as in `fetch_ml_models.ts`?

Reply:
Added a comment in `fetch_ml_models.ts`. Note the remapping will be unnecessary once we consolidate all model fetching processes in this flyout to use `fetch_ml_models.ts`.